chrome/src/background.ts

```diff
@@ -75,72 +63,93 @@ chrome.webNavigation.onBeforeNavigate.addListener(
 
 chrome.webRequest.onHeadersReceived.addListener(
   (info) => {
-    if (info.responseHeaders) {
-      const setCookieHeaders = info.responseHeaders
-        .filter(
-          (h): h is RequireFields<chrome.webRequest.HttpHeaders[0], 'value'> =>
-            h.name.toLowerCase() === 'set-cookie' &&
-            typeof h.value === 'string',
-        )
-        .map((h) => h.value)
-      if (setCookieHeaders.length === 0) {
-        return
+    if (info.parentFrameId < 0) {
+      return
+    }
+
+    // check the parent frame so we only override cookies if we are on nordcraft.com
+    nordcraftIsParentFrame({
```

Comment:
Is it possible to make the event listener asynchronous on line 65 so you can use `await` here?

Reply:
Nope, I wasn't able to do it async so that's why I did it like this.